src/build_database.py

```
@@ -138,22 +145,108 @@ def _fetch_race_ids_from_month(self, year: int, month: int) -> List[str]:
         
         return race_ids
     
-    def _scrape_race_results(self, race_id: str) -> Optional[pd.DataFrame]:
+    def _scrape_race_metadata(self, soup: BeautifulSoup) -> dict:
```

Pylint (3.3.7)

[refactor] Too many local variables (17/15)

(R0914)